### PR TITLE
Shopping cart hw

### PR DESCRIPTION
**Pull request by annton-shakur**

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstoreproject/controller/ShoppingCartController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package mate.academy.onlinebookstoreproject.controller;
```
Comment (BolsunovaNataliia) [inline src/main/java/mate/academy/onlinebookstoreproject/controller/ShoppingCartController.java:1]: Good job! But I have a few comments)

File: src/main/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepository.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import mate.academy.onlinebookstoreproject.model.ShoppingCart;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface ShoppingCartRepository extends JpaRepository<ShoppingCart, Long> {
+
+    @EntityGraph(attributePaths = {"cartItems", "cartItems.book"})
+    Optional<ShoppingCart> findByUserId(Long userId);
```
Comment (BolsunovaNataliia) [inline src/main/java/mate/academy/onlinebookstoreproject/repository/ShoppingCartRepository.java:13]: You can change -> 
Optional<ShoppingCart> findByUserId(@PathVariable(name = "id") Long userId);

File: src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        - createTable:
+            tableName: shopping_carts
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml]: You can delete nullable: false. It's redundant.

File: src/main/resources/db/changelog/changes/11-create-cart-items-table.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        - createTable:
+            tableName: cart_items
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/11-create-cart-items-table.yaml]: You can delete nullable: false. It's redundant.

**Review by BolsunovaNataliia — APPROVED** [review]

File: src/main/java/mate/academy/onlinebookstoreproject/dto/cart/AddToCartRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.onlinebookstoreproject.dto.cart;
+
+import jakarta.validation.constraints.Min;
+import lombok.Data;
+
+@Data
+public class AddToCartRequestDto {
+    private Long bookId;
+    @Min(value = 1, message = "cannot be less than 1")
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstoreproject/dto/cart/AddToCartRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/mate/academy/onlinebookstoreproject/dto/cart/CartItemUpdateDto.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package mate.academy.onlinebookstoreproject.dto.cart;
+
+import jakarta.validation.constraints.Min;
+import lombok.Data;
+
+@Data
+public class CartItemUpdateDto {
+    @Min(0)
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstoreproject/dto/cart/CartItemUpdateDto.java]:
```suggestion
    @Positive
```

File: src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml (hunk 3 above)
Comment (d1sam) [reply src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml]: agree, but it's up to you :)

File: src/main/resources/db/changelog/changes/11-create-cart-items-table.yaml (hunk 4 above)
Comment (d1sam) [reply src/main/resources/db/changelog/changes/11-create-cart-items-table.yaml]: agree, but it's up to you :) it's minor

**Review by d1sam — CHANGES_REQUESTED** [review]
good job, but there are some minor changes in dto validation need to be changed.

**Review by d1sam — APPROVED** [review]
well done!

**Pull request merged**